### ISSUE TITLE
kernel実装について

### ISSUE THREAD

**Issue by shigeki** [shigeki/ask_nishida_about_quic_jp#5]
>WebExで、さかねさんから下記の質問/コメントが入っていました。
>
>-----

Shoichi Sakane から 全員 へ:
将来的にはkernalに入ってこないんですか？仕様で

Shoichi Sakane から 全員 へ:
ユーザランドに実装しろと言われてるわけではないですよね？

Shoichi Sakane から 全員 へ:
きっとTSURUプロジェクトがカーネルに突っ込みますよ

**Comment by shigeki** [comment]
仕様的に userland に限定することはないと思います。QUICはTCPと違ってバージョンネゴシエーションの機能を備えているので頻繁なバージョンアップで機能の追加や見直しを行うことが期待されています。そのため当初は userland での実装が主流になるだろうと予想しています。

QUIC を kernel 実装にするためには TLS1.3の機能を実装する必要があるため、暗号や認証系の実装の整備が必要になるでしょう。Linuxでは今見たら AES-GCMやChaCha20-Poly1305, SHA256など用意されていますが、RSA,E CDSA, ECDHE, X.509やそれに伴うASN.1処理系の実装などが必要になるのではと思います。

**Comment by nsd246** [comment]
はい。Seoulのミーティングでも同様の質問が出ていましたが、draft authorがユーザランドに限定しているわけではないという返答をしていたと思います。
